main/src/cgeo/geocaching/utils/BackupUtils.java

```diff
@@ -380,99 +407,101 @@ private File createSettingsBackupInternal(final File backupDir, final Boolean fu
+            msg = activityContext.getString(R.string.backup_saved) + "\n" + backupDir.toUserDisplayableString();
         } else {
             title = activityContext.getString(R.string.init_backup_backup_failed);
 
-            if (databaseFile != null) {
-                msg = activityContext.getString(R.string.init_backup_success) + "\n" + databaseFile;
+            if (databaseResult != null) {
+                msg = activityContext.getString(R.string.init_backup_success) + "\n" + backupDir.toUserDisplayableString() + "/" + DataStore.DB_FILE_NAME_BACKUP;
             } else {
                 msg = activityContext.getString(R.string.init_backup_failed);
             }
 
             msg += "\n\n";
 
-            if (settingsFile != null) {
-                msg += activityContext.getString(R.string.settings_saved) + "\n" + settingsFile;
+            if (settingsResult != null) {
+                msg += activityContext.getString(R.string.settings_saved) + "\n" + backupDir.toUserDisplayableString() + "/" + SETTINGS_FILENAME;
             } else {
                 msg += activityContext.getString(R.string.settings_savingerror);
             }
         }
 
+        final ArrayList<Uri> files = new ArrayList<>();
+
+        for (ContentStorage.FileInformation fi : ContentStorage.get().list(backupDir)) {
+            files.add(fi.uri);
+        }
+
         Dialogs.messageNeutral(activityContext, title, msg, R.string.cache_share_field,
-            (dialog, which) -> ShareUtils.shareMultipleFiles(activityContext, Arrays.asList(settingsFile, databaseFile), R.string.init_backup_backup));
+            (dialog, which) -> ShareUtils.shareMultipleFiles(activityContext, files, R.string.init_backup_backup));
     }
 
 
-    /**
-     * Public methods for checking the backup availability
-     */
+    /* Methods for checking the backup availability */
+
+    public static boolean hasBackup(final Folder backupDir) {
+        // TODO:
+        //  When selecting a user defined restore directory, ContentStorage.get().list(backupDir) returns an empty List although files are present,
+        //  so getDatabaseFile()/getSettingsFile() returns null.
+        //     List<ContentStorage.FileInformation> a = ContentStorage.get().list(backupDir);
+        //  Why is it not working?
 
-    public static boolean hasBackup(final File backupDir) {
         return getDatabaseFile(backupDir) != null || getSettingsFile(backupDir) != null;
     }
 
     @Nullable
-    public static File getDatabaseFile(final File backupDir) {
-        final File fileSourceFile = DataStore.getBackupFileInternal(backupDir, true);
-        return fileSourceFile.exists() && fileSourceFile.length() > 0 ? fileSourceFile : null;
+    private static ContentStorage.FileInformation getDatabaseFile(final Folder backupDir) {
+        return ContentStorage.get().getFileInfo(backupDir, DataStore.DB_FILE_NAME_BACKUP);
```

Comment:
As `ContentStorage.get().getFileInfo` uses `list(Folder)` internally, the file is not found...